src/test/java/spoon/test/factory/CodeFactoryTest.java

```diff
@@ -66,4 +68,54 @@ void createCtCatcVariablehWithoutModifiers() {
 		CtTypeReference<Exception> exceptionType = factory.Type().createReference(Exception.class);
 		assertDoesNotThrow(() -> factory.Code().createCatchVariable(exceptionType, "e"));
 	}
+
+	@ParameterizedTest
+	@ValueSource(classes = {
+		byte[].class,
+		byte[][].class,
+		String[][].class,
+	})
+	void testCreateCtReferenceOfArray(Class<?> clazz) {
+		// contract: References for arrays are created
+		Factory factory = createFactory();
+		assertEquals(
+			clazz.getSimpleName(),
+			factory.createCtTypeReference(clazz).getSimpleName()
+		);
+	}
+
+	@Test
+	void testCreateCtReferenceOfInnerClass() {
+		// contract: References for inner classes are created
+		Factory factory = createFactory();
+		assertEquals(
+			TestClassInnerClass.class.getSimpleName(),
+			factory.createCtTypeReference(TestClassInnerClass.class).getSimpleName()
+		);
+	}
+
+	@Test
+	void testCreateCtReferenceOfLocalClass() {
+		class LocalClass {}
+		// contract: References for local classes are created
+		Factory factory = createFactory();
+		assertEquals(
+			LocalClass.class.getSimpleName(),
+			factory.createCtTypeReference(LocalClass.class).getSimpleName()
+		);
+	}
+
+	@Test
+	void testCreateCtReferenceOfAnonymousClass() {
+		// contract: References for anonymous classes are created
+		Class<?> clazz = (new Object() {
+		}.getClass());
+		Factory factory = createFactory();
+		assertEquals(
+			clazz.getName(),
+			factory.createCtTypeReference(clazz).getQualifiedName()
+		);
+	}
+
+	private static class TestClassInnerClass {}
```

Comment:
Maybe move this test object closer to the usage?

Reply:
I inlined it.